trace.go

```diff
@@ -76,18 +76,20 @@ func (f *firstLine) String() string {
 	return line.String()
 }
 
-// payload represents an RPC request or response payload.
-type payload struct {
-	sent bool        // whether this is an outgoing payload
-	msg  interface{} // e.g. a proto.Message
-	// TODO(dsymonds): add stringifying info to codec, and limit how much we hold here?
+const truncateSize = 100
+
+func truncate(x string, l int) string {
+	if l > len(x) {
+		return x
+	}
+	return string([]byte(x[:l]))
 }
 
-func (p payload) String() string {
-	if p.sent {
-		return fmt.Sprintf("sent: %v", p.msg)
+func newPayload(sent bool, msg interface{}) stringer {
+	if sent {
+		return stringer(truncate(fmt.Sprintf("sent: %v", msg), truncateSize))
```

Comment:
For huge amounts of data and msgs that are natively `Stringers`, it's probably best to call `truncate(stringerMsg.String(), truncateSize)` inside `fmt.Sprintf()` to avoid an allocation and copy of the data that will be removed anyway.  But this is _probably_ a pretty trivial savings.